### PR TITLE
Fix another bunch of HTML violations

### PR DESCRIPTION
**Pull request by RoiEXLab**
There are still a lot of semantical errors, but this PR should at least fix all syntactical errors I found (related to escaping).
Also a regression is being fixed I introduced in #347 : I changed a div to a span, which contains non-inline content being inserted by nodebb. Solution was to move this span out of the surrounding h2 tag and change it back to a div.
For some reason the "subtopics image" shrunk by a couple of pixels during this process, everything else is almost exactly the same:
Before:
Before
After: After
If that's a problem, and you could tell me how to fix it (I couldn't figure it out), I will do so.

### CODE REVIEW

File: less/category.less [hunk 1]
```diff
@@ -130,13 +130,24 @@
 		color: @gray-light;
 	}
 
-	h2.title .description {
-		font-size: 65%;
+	.description {
+		font-size: 11.7px;
```
Comment (julianlam) [inline less/category.less]: Instead of a pixel value, can you use `rem`s?

File: templates/topic.tpl [hunk 2]
```diff
@@ -34,7 +34,7 @@
 		<ul component="topic" class="posts" data-tid="{tid}" data-cid="{cid}">
 			<!-- BEGIN posts -->
 				<li component="post" class="<!-- IF posts.deleted -->deleted<!-- ENDIF posts.deleted -->" <!-- IMPORT partials/data/topic.tpl -->>
-					<a component="post/anchor" data-index="{posts.index}" name="{posts.index}"></a>
+					<a component="post/anchor" data-index="{posts.index}" id="{posts.index}"></a>
```
Comment (julianlam) [inline templates/topic.tpl]: I think `id` is necessary for inter-page traversal via anchors?

Comment (RoiEXLab) [reply templates/topic.tpl]: I'm confused...
I replaced name with id, if you mean it the other way round then I just can say, that name is a "legacy reserved" keyword for all elements, and I haven't found any reference to this attribute in the nodebb core code.

File: less/category.less (hunk 1 above)
Comment (RoiEXLab) [reply less/category.less]: Sure

File: templates/topic.tpl (hunk 2 above)
Comment (pitaj) [reply templates/topic.tpl]: Both `id` and `name` work with anchor traversal, @julianlam

**Comment by RoiEXLab** [comment]
Bumping @julianlam

**Comment by julianlam** [comment]
No objections, will merge after v1.7.0

**Comment by RoiEXLab** [comment]
Bumping @julianlam

**Comment by julianlam** [comment]
:+1:

File: less/category.less [hunk 3]
```diff
@@ -130,13 +130,24 @@
 		color: @gray-light;
 	}
 
-	h2.title .description {
-		font-size: 65%;
+	.description {
+		font-size: 0.73125rem;
```
Comment (julianlam) [inline less/category.less]: Why did this change get added? It makes category descriptions really tiny, so we've reverted it now.

Comment (RoiEXLab) [reply less/category.less]: I wanted to prevserve the size...
Unwanted side effect, sorry about that

**Pull request merged**